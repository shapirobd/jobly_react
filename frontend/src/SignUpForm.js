import React from "react";

const SignUpForm = ({ handleChange, handleSubmit, errorMessage }) => {
	return (
		<form className="SignUp-form" onSubmit={handleSubmit}>
			<label htmlFor="username">Username</label>
			<input
				type="text"
				onChange={handleChange}
				name="username"
				id="username-input"
			></input>
			{errorMessage ? <p className="SignUp-form-err">{errorMessage}</p> : null}
			<label htmlFor="password">Password</label>
			<input
				type="password"
				onChange={handleChange}
				name="password"
				id="password-input"
			></input>
			<label htmlFor="firstName">First Name</label>
			<input
				type="text"
				onChange={handleChange}
				name="firstName"
				id="firstName-input"
			></input>
			<label htmlFor="lastName">Last Name</label>
			<input
				type="text"
				onChange={handleChange}
				name="lastName"
				id="lastName-input"
			></input>
			<label htmlFor="email">Email</label>
			<input
				type="email"
				onChange={handleChange}
				name="email"
				id="email-input"
			></input>
			<div className="SignUp-btn-wrapper">
				<button type="submit" className="SignUp-btn">
					Submit
				</button>
			</div>
		</form>
	);
};

export default SignUpForm;
