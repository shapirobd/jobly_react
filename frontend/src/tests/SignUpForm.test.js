import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignUpForm from "../SignUpForm";

test("smoke test", () => {
	render(
		<MemoryRouter>
			<SignUpForm />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<SignUpForm />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
