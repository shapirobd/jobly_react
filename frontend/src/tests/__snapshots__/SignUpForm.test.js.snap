// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <form
    class="SignUp-form"
  >
    <label
      for="username"
    >
      Username
    </label>
    <input
      id="username-input"
      name="username"
      type="text"
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password-input"
      name="password"
      type="password"
    />
    <label
      for="firstName"
    >
      First Name
    </label>
    <input
      id="firstName-input"
      name="firstName"
      type="text"
    />
    <label
      for="lastName"
    >
      Last Name
    </label>
    <input
      id="lastName-input"
      name="lastName"
      type="text"
    />
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email-input"
      name="email"
      type="email"
    />
    <div
      class="SignUp-btn-wrapper"
    >
      <button
        class="SignUp-btn"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
